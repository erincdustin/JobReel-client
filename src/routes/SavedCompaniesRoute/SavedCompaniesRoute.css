.SavedCompaniesRoute {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 1em;
}

.savedCompBack {
  position: relative;
  grid-column-start: 1;
  grid-column-end: 9;
}

.savedCompBack > img {
  height: 200px;
  width: 200px;
  position: absolute;
  top: -170px;
  right: 0%;
  z-index: -1;
}

@media (max-width: 960px) {
  .SideNav{
    display: none;
  }
}

@media (min-width: 960px) {
  .TopNav{
    display: none;
  }
}