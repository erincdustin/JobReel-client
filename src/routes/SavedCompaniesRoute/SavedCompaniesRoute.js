import React from 'react';
import SideNav from '../../components/SideNav/SideNav';
import AddCompanyForm from '../../components/AddCompanyForm/AddCompanyForm';
import jobReelApiService from '../../services/jobreel-api-service';
import JobReelContext from '../../context/JobReelContext';
import CompanySummary from '../../components/CompanySummary/CompanySummary';
import MediaQuery from 'react-responsive';
import savedCompBack from '../../assests/savedCompBack.svg'
import TopNav from '../../components/TopNav/TopNav'
import './SavedCompaniesRoute.css';

class SavedCompaniesRoute extends React.Component {
  
  static contextType = JobReelContext

  componentDidMount = () => {
    if (this.context.companies.length === 0)
    jobReelApiService.getSavedCompanies()
      .then(res => {
        this.context.setCompanies(res);
      })
  }

  render() {
    let display;
    (this.context.manualCompanyAdd === false) ? display = <CompanySummary /> : display = <AddCompanyForm /> 
    return (
      <div className="SavedCompaniesRoute">
        <div className='title'>
          <h2>Saved Companies</h2>
        </div>
        <div className='savedCompBack'>
          <img src={savedCompBack} alt='saved-company-bakcground'/>
        </div>
        <MediaQuery minDeviceWidth={961}>
          <SideNav/>
        </MediaQuery>
        <MediaQuery maxDeviceWidth={960}>
          <TopNav/>
        </MediaQuery>
        {display}
      </div>
    )
  }
}

export default SavedCompaniesRoute;