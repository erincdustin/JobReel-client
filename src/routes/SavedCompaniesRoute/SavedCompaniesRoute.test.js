import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import SavedCompaniesRoute from './SavedCompaniesRoute'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PropTypes from 'prop-types';

class ContextProvider extends React.Component {
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      companies: [1, 2, 3]
    }
  })

  render() {
    return this.props.children;
  }
}

describe('SavedCompaniesRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><SavedCompaniesRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('SavedCompaniesRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
      <ContextProvider>
        <SavedCompaniesRoute />
      </ContextProvider>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})