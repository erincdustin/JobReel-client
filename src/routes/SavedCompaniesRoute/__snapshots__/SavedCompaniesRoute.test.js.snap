// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedCompaniesRoute snapshot test renders the UI as expected 1`] = `<SavedCompaniesRoute />`;
